apps/condo/domains/organization/schema/ResetOrganizationService.js

```diff
@@ -0,0 +1,184 @@
+const { loadListByChunks } = require('@condo/domains/common/utils/serverSchema')
+const { B2BAppContext } = require('@condo/domains/miniapp/utils/serverSchema')
+const access = require('@condo/domains/organization/access/ResetOrganizationService')
+const { DELETED_ORGANIZATION_NAME } = require('@condo/domains/organization/constants/common')
+const { Organization, OrganizationLink, OrganizationEmployee } = require('@condo/domains/organization/utils/serverSchema')
+const { Property } = require('@condo/domains/property/utils/serverSchema')
+
+/**
+ * List of possible errors, that this custom schema can throw
+ * They will be rendered in documentation section in GraphiQL for this custom schema
+ */
+const errors = {
+    DV_VERSION_MISMATCH: {
+        mutation: 'resetOrganization',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Unsupported value for dv',
+    },
+    SENDER_MISMATCH: {
+        mutation: 'resetOrganization',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Unsupported value for sender',
+    },
+    ORGANIZATION_NOT_FOUND: {
+        mutation: 'resetOrganization',
+        variable: ['data', 'organizationId'],
+        code: BAD_USER_INPUT,
+        type: 'ORGANIZATION_NOT_FOUND',
+        message: 'Could not find organization by provided id',
+    },
+}
+
+const ResetOrganizationService = new GQLCustomSchema('ResetOrganizationService', {
+    types: [
+        {
+            access: true,
+            type: 'input ResetOrganizationInput { dv: Int! sender: JSON! organizationId: ID! }',
+        },
+        {
+            access: true,
+            type: 'type ResetOrganizationOutput { status: String! }',
+        },
+    ],
+    
+    mutations: [
+        {
+            access: access.canResetOrganization,
+            schema: 'resetOrganization(data: ResetOrganizationInput!): ResetOrganizationOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data: { dv, sender, organizationId } } = args
+                checkDvAndSender({ dv, sender }, { ...errors.DV_VERSION_MISMATCH, mutation: 'resetOrganization' }, { ...errors.SENDER_MISMATCH, mutation: 'resetOrganization' }, context)
+
+                const DV_SENDER = { dv, sender }
+
+                if (!organizationId) throw new Error('resetOrganization(): no organizationId')
+
+                const organizationEntity = await getById('Organization', organizationId)
+                if (!organizationEntity) {
+                    throw new GQLError(errors.ORGANIZATION_NOT_FOUND, context)
+                }
+
+                const properties = await loadListByChunks({
+                    context,
+                    list: Property,
+                    chunkSize: 20,
+                    limit: 100000,
```

Comment:
looks like lots of very potentially heavy operations. can we move all these logic to async task ?

Reply:
We use to move heavy calculations/requests to async tasks in order not to block API for too long.